src/sync/streaming/pushManager.ts

```diff
@@ -221,7 +221,17 @@ export function pushManagerFactory(
   /** Functions related to synchronization (Queues and Workers in the spec) */
 
   pushEmitter.on(SPLIT_KILL, splitsUpdateWorker.killSplit);
-  pushEmitter.on(SPLIT_UPDATE, splitsUpdateWorker.put);
+  pushEmitter.on(SPLIT_UPDATE, (parsedData) => {
+    if (parsedData.d !== undefined) {
+      const payload = parseFFUpdatePayload(parsedData);
```

Comment:
You must try-catch the call to `parseFFUpdatePayload`. In case of exception (e.g., error decoding base64 or decompressing) it should follow the original flow of calling `splitsUpdateWorker.put(parsedData);`